### ISSUE TITLE
Unable to get renv to install packages on Ubuntu 24.04

### ISSUE THREAD

**Issue by cramjaco** [rstudio/renv#1917]
Dear renv developers. I'm struggling to install any packages under renv. 
I just posted to posit support here with full details of my issue:

https://forum.posit.co/t/cannot-install-packages-with-renv/187450

In brief, I can install without renv, no packages can be found if I activate renv and try to install packages, many troubleshooting steps  (see link) had no effect.

And would love any guidance anyone here could provide.

**Comment by kevinushey** [comment]
```
> Diagnostics Report [renv 0.16.0]
> ================================
```

You're using a rather old version of `renv` -- are you able to try updating to the latest version and trying again?

The documentation at https://rstudio.github.io/renv/articles/package-install.html#downloads may also be useful.

**Comment by cramjaco** [comment]
Thanks for the suggestion, and I'm not sure why the updated version didn't stick. I updated to renv 1.0.7 with details in the linked forum post. Still no luck.

Following the documentation that kevinushey provided, which I have seen before, I found by running options(renv.download.override = utils::download.file) I managed to revert `getOption("download.file.method")` to returning null. The other options don't seem to cause any improvement as far as I can tell. The other sections besides Alternative Downloaders seem less relevant. 

So in conclusion, I'm still stuck. Thanks for any other advice or suggestions.

**Comment by kevinushey** [comment]
Are you still getting this same warning / error?

> could not retrieve available packages for url 'https://cloud.r-project.org/src/contrib' 

Could you paste the full command output + error output for posterity as well? Are you using the same repositories in each scenario?

**Comment by cramjaco** [comment]
Sure thing.
With a restart of R, I'm getting the following now:

```
> install.packages("here")
renv was unable to query available packages from the following repositories:
- # https://packagemanager.posit.co/cran/__linux__/noble/latest/src/contrib --------
error downloading 'https://packagemanager.posit.co/cran/__linux__/noble/latest/src/contrib/PACKAGES.rds' [error code 26]
error downloading 'https://packagemanager.posit.co/cran/__linux__/noble/latest/src/contrib/PACKAGES.gz' [error code 26]
error downloading 'https://packagemanager.posit.co/cran/__linux__/noble/latest/src/contrib/PACKAGES' [error code 26]


Error: package 'here' is not available
Traceback (most recent calls last):
15: install.packages("here")
14: eval(call, envir = parent.frame())
13: eval(call, envir = parent.frame())
12: renv::install("here")
11: retrieve(packages)
10: handler(package, renv_retrieve_impl(package))
 9: renv_retrieve_impl(package)
 8: withCallingHandlers(renv_available_packages_latest(package), 
        error = function(err) stopf("package '%s' is not available", 
            package))
 7: renv_available_packages_latest(package)
 6: stopf("package '%s' is not available", package)
 5: stop(sprintf(fmt, ...), call. = call.)
 4: .handleSimpleError(function (err) 
    {
        ...
    }, "package 'here' is not available", base::quote(NULL))
 3: h(simpleError(msg, call))
 2: stopf("package '%s' is not available", package)
 1: stop(sprintf(fmt, ...), call. = call.)
```

Let me know if I can run anything else!

**Comment by kevinushey** [comment]
Can you try once more with:

```
options(renv.download.trace = TRUE)
```

And then perhaps:

```
renv::install("here", repos = "https://cran.rstudio.com")
```

Or, alternatively:

```
Sys.setenv(RENV_DOWNLOAD_METHOD = "libcurl")
renv::install("here")
```

The fact that you're getting error code 26 from curl makes me think the issue might be with attempting to download files into the R temporary directory as well.

**Comment by kevinushey** [comment]
Great! You might want to set:

```
RENV_DOWNLOAD_METHOD = libcurl
```

in your project's `.Renviron` file, so that `renv` will know to use that download method by default.

**Comment by cramjaco** [comment]
I'm reopening this issue.
I find that the fix only worked when I ran this in a new directory in which I had forgotten to run `renv::init()`

Thus, if I run
> renv::init()
> Sys.setenv(RENV_DOWNLOAD_METHOD = "libcurl")
> renv::install("here")
Error: package 'here' is not available

So still not working from my perspective.

**Comment by cramjaco** [comment]
@kevinushey I realize I don't have permission to re-open this issue. Following my comment above, can you please do so?
Thanks.
-Jacob

**Comment by cramjaco** [comment]
I'm writing to confirm that this problem appears to persist with R version 4.4.1 and renv version 1.0.11
renv appears to be unusable still for ubuntu 24.04. If anyone manages to get renv working on this version I'd be interested to hear how.
